agent/stats/service_connect_linux_test.go

```diff
@@ -121,31 +121,33 @@ func TestRetrieveServiceConnectMetrics(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		// Set up a mock http sever on the statsUrlpath
-		mockUDSPath := "/tmp/appnet_admin.sock"
-		r := mux.NewRouter()
-		r.HandleFunc("/get/them/stats", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			fmt.Fprintf(w, "%v", test.stats)
-		}))
+		func() {
+			// Set up a mock http sever on the statsUrlpath
+			mockUDSPath := "/tmp/appnet_admin.sock"
+			r := mux.NewRouter()
+			r.HandleFunc("/get/them/stats", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				fmt.Fprintf(w, "%v", test.stats)
+			}))
 
-		ts := httptest.NewUnstartedServer(r)
+			ts := httptest.NewUnstartedServer(r)
+			defer ts.Close()
```

Comment:
I received this error when running unit test `listen unix /tmp/appnet_admin.sock: bind: address already in use`
Moving `ts.Close()` up in a defer statement should help us clean up always.